pkg/odo/cli/url/create.go

```diff
@@ -182,10 +224,11 @@ func NewCmdURLCreate(name, fullName string) *cobra.Command {
 	}
 	urlCreateCmd.Flags().IntVarP(&o.urlPort, "port", "", -1, "port number for the url of the component, required in case of components which expose more than one service port")
 	urlCreateCmd.Flags().BoolVarP(&o.secureURL, "secure", "", false, "creates a secure https url")
-	// if experimental mode is enabled, add more flags to support ingress creation
+	// if experimental mode is enabled, add more flags to support ingress creation based on devfile
 	if experimental.IsExperimentalModeEnabled() {
-		urlCreateCmd.Flags().StringVar(&o.tlsSecret, "tlsSecret", "", "tls secret name for the url of the component if the user bring his own tls secret")
-		urlCreateCmd.Flags().StringVar(&o.clusterHost, "clusterHost", "", "Cluster host for this URL")
+		urlCreateCmd.Flags().StringVar(&o.tlsSecret, "tls-secret", "", "tls secret name for the url of the component if the user bring his own tls secret")
+		urlCreateCmd.Flags().StringVarP(&o.host, "host", "", "", "Cluster ip for this URL")
+		urlCreateCmd.Flags().StringVar(&o.DevfilePath, "devfile", "./devfile.yaml", "Path to a devfile.yaml")
```

Comment:
This seems to use the Devfile path from https://github.com/openshift/odo/pull/2762/files#diff-6e3a2602ffd394e9a73d4fbf1388be8dR40. I think we should have it in URLCreateOptions itself.

Reply:
I have raised up the same concern during our team meeting, but the decision is that we do not save the devfilePath into `env.yaml` during create. 
Since the default devfile path should be `./devfile.yaml`, if the user brings their own, it is also encourage to put it under the project directory. They need to specify the path if they put that file elsewhere.

Reply:
I posted this concern again in the teem, and waiting for other's thoughts. 
I still think saving devfilePath in `env.yaml` during creation makes more sense. Since create cmd is required before other commands being executed. So for any following command (push, url create, etc.), we can directly get it from the env.yaml file. 

If the user wants to use another devfilePath(i.e. change the value of devfilePath), we can also allow those commands which accept the `--devfile` to overwrite the value within `env.yaml`.
@elsony @kadel @johnmcollier Any thought?

(But anyways, if we are going to do that, I would prefer to have a separate issue and handle that in a separate PR. I would say leave the `DevfilePath` exposed as it is for now.)

Reply:
OK